packages/components/src/internal/components/administration/PermissionManagementPage.tsx

```diff
@@ -2,37 +2,95 @@
+import { AUDIT_EVENT_TYPE_PARAM, GROUP_AUDIT_QUERY, PROJECT_AUDIT_QUERY } from '../auditlog/constants';
+
+import { BasePermissionsCheckPage } from '../permissions/BasePermissionsCheckPage';
+import { PermissionAssignments } from '../permissions/PermissionAssignments';
+import { InjectedRouteLeaveProps, withRouteLeave } from '../../util/RouteLeave';
+import { InjectedPermissionsPage, withPermissionsPage } from '../permissions/withPermissionsPage';
+
 import { useAdminAppContext } from './useAdminAppContext';
+import { APPLICATION_SECURITY_ROLES, HOSTED_APPLICATION_SECURITY_ROLES, SITE_SECURITY_ROLES } from './constants';
+import { showPremiumFeatures } from './utils';
+
+// exported for testing
+export type Props = InjectedRouteLeaveProps & InjectedPermissionsPage & WithRouterProps;
 
-export const PermissionManagementPage: FC = memo(() => {
-    const { container, moduleContext, user } = useServerContext();
+// exported for testing
+export const PermissionManagementPageImpl: FC<Props> = memo(props => {
+    const { children } = props;
+
+    const [policyLastModified, setPolicyLastModified] = useState<string>(undefined);
+    const [hidePageDescription, setHidePageDescription] = useState<boolean>(false);
+    const { dismissNotifications, createNotification } = useNotificationsContext();
     const { extraPermissionRoles } = useAdminAppContext();
+    const { user, moduleContext, container } = useServerContext();
+    const showAssignments = user.isAdmin || user.isRootAdmin;
     const showPremium = showPremiumFeatures(moduleContext);
+    const description = showPremium ? container.path : undefined;
+
+    const onSuccess = useCallback(() => {
+        dismissNotifications();
+        createNotification('Successfully updated roles and assignments.');
+    }, [createNotification, dismissNotifications]);
+
+    const renderButtons = useCallback(() => {
+        const row = policyLastModified ? { Modified: { value: policyLastModified } } : {};
 
-    const rolesMap = useMemo(() => {
-        let roles = showPremium ? APPLICATION_SECURITY_ROLES : HOSTED_APPLICATION_SECURITY_ROLES;
-        roles = roles.merge(Map<string, string>(extraPermissionRoles));
-        return roles;
+        return (
+            <>
+                <CreatedModified row={row} />
+                <a href={AppURL.create(AUDIT_KEY).addParam(AUDIT_EVENT_TYPE_PARAM, GROUP_AUDIT_QUERY.value).toHref()}>
+                    View Audit History
+                </a>
+            </>
+        );
+    }, [policyLastModified]);
+
+    const rolesToShow = useMemo(() => {
+        let roles_ = showPremium ? APPLICATION_SECURITY_ROLES : HOSTED_APPLICATION_SECURITY_ROLES;
+        roles_ = roles_.merge(Map<string, string>(extraPermissionRoles));
+        return roles_.keySeq().toList();
     }, [extraPermissionRoles, showPremium]);
 
+    const setProjectCount = useCallback((projectCount: number) => {
+        setHidePageDescription(projectCount > 1);
```

Comment:
For LKS premium users we probably want to show folder path because they might have projects/folders outside the application. My logic is that if there is only 1 folder (which will be the Home folder), then show path for premium users. Otherwise, because the project list on the left highlights the folder that they are changing permission for, there is no need to show folder path separately.